### PR TITLE
Add new test for PodTemplates

### PR DESCRIPTION
**Pull request by BobyMCbobs**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Add new test for PodTemplate creation

**Special notes for your reviewer**:
Test created as per comments from @spiffxp, found [here](https://github.com/kubernetes/kubernetes/pull/79871#issuecomment-514854868)

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

/area test
@kubernetes/sig-apimachinery-pr-reviews
@kubernetes/sig-architecture-pr-reviews

### CODE REVIEW

File: test/e2e/common/pods.go [hunk 1]
```diff
@@ -833,4 +833,26 @@ var _ = framework.KubeDescribe("Pods", func() {
...
+						Containers: []v1.Container{
+							{Name: "test", Image: imageutils.GetE2EImage(imageutils.Agnhost)},
+						},
+					},
+				},
+			})
+		framework.ExpectNoError(err)
+
+		ginkgo.By("validating the creation of the PodTemplate")
+		_, err = f.ClientSet.CoreV1().PodTemplates(f.Namespace.Name).Get("pod-template-test", metav1.GetOptions{})
```
Comment (spiffxp) [inline test/e2e/common/pods.go]: this only tests that the Get call won't error, you probably want to verify that what you get back is a pod template, and that its contents match what you expect given what you created above

**Review by spiffxp — CHANGES_REQUESTED** [review]
this is a good start but doesn't feel like enough of a test for pure CRUD semantics

if we're looking to test more than CRUD, I'm not actually sure what behaviors are expected of a pod template on its own?

**Comment by bgrant0607** [comment]
Does anyone use the PodTemplate API? It was created for #170, which was never implemented. I'd bet a lot of clusters disable this API or set ResourceQuota to 0 so that it's not a DoS vector.

This should NOT be added to conformance.

cc @johnbelamaric

**Comment by liggitt** [comment]
> Does anyone use the PodTemplate API? It was created for #170, which was never implemented.

I have never seen it used anywhere.

**Comment by johnbelamaric** [comment]
Can it be deprecated?

**Comment by devaips** [comment]
@BobyMCbobs let's close this PR on account of the PodTemplate API being unused/on-the-way-to-deprecated.

**Comment by BobyMCbobs** [comment]
> @BobyMCbobs let's close this PR on account of the PodTemplate API being unused/on-the-way-to-deprecated.

Closing because of deprecation.
/close

**Pull request closed without merging**